delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+                        'resource_type': constants.DEFAULT_RESOURCE_TYPE,
+                        'location': ''
+                    }
+                    return alert_model
+        except exception.DelfinException as e:
+            err_msg = "Failed to parse alert from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to parse alert from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def list_ports(self, storage_id):
+        try:
+            ports_list = []
+            ports_list.extend(self.get_fc_port(storage_id))
+            ports_list.extend(self.get_eth_port(storage_id))
+            return ports_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get ports from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get ports from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_fc_port(self, storage_id):
+        try:
+            fc_info = self.ssh_do_exec([constant.FC_PORT_COMMAND])
+            fc_map_list = []
+            self.split_value_map_list(fc_info, fc_map_list, 'Portname')
+            fc_list = []
+            speed_info = self.ssh_do_exec([constant.FC_SPEED_COMMAND])
+            speed_map_list = []
+            self.split_value_map_list(speed_info, speed_map_list, 'FC1')
+            speed_map = speed_map_list[-1]
+            for value_map in fc_map_list:
+                if 'Portname' in value_map:
+                    status = value_map.get('Status', None)
+                    health = constants.PortHealthStatus.ABNORMAL
+                    if status and status == 'Good':
+                        health = constants.PortHealthStatus.NORMAL
+                    connection_status = \
+                        constants.PortConnectionStatus.DISCONNECTED
+                    if 'FCLinkisup' in value_map:
+                        connection_status = \
+                            constants.PortConnectionStatus.CONNECTED
+                    port_id = ''
+                    for key in value_map.keys():
+                        if 'HostPort' in key:
+                            port_id = key.replace('HostPort', '')
+                            break
+                    speed = \
+                        int(speed_map.get('FC' + port_id).replace('Gbps', ''))
+                    fc_model = {
+                        'name': 'FC' + port_id,
+                        'storage_id': storage_id,
+                        'native_port_id': port_id,
+                        'location': None,
```

Comment:
If info cannot be attained from storage, it can be deleted, and same as 7 lines below.

Reply:
has  been modify